zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -0,0 +1,565 @@
+            Self::SMOD,
+            Self::ADDMOD,
+            Self::MULMOD,
+            Self::EXP,
+            Self::SIGNEXTEND,
+            Self::CMP,
+            Self::SCMP,
+            Self::ISZERO,
+            Self::BITWISE,
+            Self::NOT,
+            Self::BYTE,
+            Self::SHL,
+            Self::SHR,
+            Self::SAR,
+            Self::SHA3,
+            Self::ADDRESS,
+            Self::BALANCE,
+            Self::ORIGIN,
+            Self::CALLER,
+            Self::CALLVALUE,
+            Self::CALLDATALOAD,
+            Self::CALLDATASIZE,
+            Self::CALLDATACOPY,
+            Self::CODESIZE,
+            Self::CODECOPY,
+            Self::GASPRICE,
+            Self::EXTCODESIZE,
+            Self::EXTCODECOPY,
+            Self::RETURNDATASIZE,
+            Self::RETURNDATACOPY,
+            Self::EXTCODEHASH,
+            Self::BLOCKHASH,
+            Self::COINBASE,
+            Self::TIMESTAMP,
+            Self::NUMBER,
+            Self::DIFFICULTY,
+            Self::GASLIMIT,
+            Self::CHAINID,
+            Self::SELFBALANCE,
+            Self::BASEFEE,
+            Self::POP,
+            Self::MEMORY,
+            Self::SLOAD,
+            Self::SSTORE,
+            Self::JUMP,
+            Self::JUMPI,
+            Self::PC,
+            Self::MSIZE,
+            Self::GAS,
+            Self::JUMPDEST,
+            Self::PUSH,
+            Self::DUP,
+            Self::SWAP,
+            Self::LOG,
+            Self::CREATE,
+            Self::CALL,
+            Self::CALLCODE,
+            Self::RETURN,
+            Self::DELEGATECALL,
+            Self::CREATE2,
+            Self::STATICCALL,
+            Self::REVERT,
+            Self::SELFDESTRUCT,
+            Self::ErrorInvalidOpcode,
+            Self::ErrorStackOverflow,
+            Self::ErrorStackUnderflow,
+            Self::ErrorWriteProtection,
+            Self::ErrorDepth,
+            Self::ErrorInsufficientBalance,
+            Self::ErrorContractAddressCollision,
+            Self::ErrorMaxCodeSizeExceeded,
+            Self::ErrorInvalidCreationCode,
+            Self::ErrorReverted,
+            Self::ErrorInvalidJump,
+            Self::ErrorReturnDataOutOfBound,
+            Self::ErrorOutOfGasConstant,
+            Self::ErrorOutOfGasPureMemory,
+            Self::ErrorOutOfGasCodeStore,
+            Self::ErrorOutOfGasSHA3,
+            Self::ErrorOutOfGasCALLDATACOPY,
+            Self::ErrorOutOfGasCODECOPY,
+            Self::ErrorOutOfGasEXTCODECOPY,
+            Self::ErrorOutOfGasRETURNDATACOPY,
+            Self::ErrorOutOfGasLOG,
+            Self::ErrorOutOfGasCALL,
+            Self::ErrorOutOfGasCALLCODE,
+            Self::ErrorOutOfGasDELEGATECALL,
+            Self::ErrorOutOfGasCREATE2,
+            Self::ErrorOutOfGasSTATICCALL,
+        ]
+        .iter()
+        .copied()
+    }
+
+    pub(crate) fn amount() -> usize {
+        Self::iterator().count()
+    }
```

Comment:
A little nitpick:
There is a crate called [strum](https://crates.io/crates/strum), it has macros `EnumIter` and `EnumCount` do the exact functions.
You can just add these two attributes, it can save a lot codes.

Reply:
Thanks! Will try it in further PR.